shell/components/Inactivity.vue

```diff
@@ -0,0 +1,203 @@
+import ModalWithCard from '@/shell/components/ModalWithCard';
+import { Banner } from '@components/Banner';
+import PercentageBar from '@/shell/components/PercentageBar.vue';
+
+export default {
+  name:       'Inactivity',
+  components: {
+    ModalWithCard, Banner, PercentageBar
+  },
+
+  props: {
+    showModalAfter: {
+      type:    Number,
+      default: 5, // in seconds
+    },
+    courtesyTimer: {
+      type:    Number,
+      default: 10, // in seconds
+    },
+    enabled: {
+      type:    Boolean,
+      default: false,
+    }
+  },
+
+  data() {
+    return {
+      isOpen:              false,
+      isInactive:          false,
+      inactivityTimeoutId: null,
+      courtesyCountdown:   this.courtesyTimer,
+      courtesyTimerId:     null,
+    };
+  },
+  mounted() {
+    if (!this.enabled) {
+      return;
+    }
+
+    this.addIdleListeners();
+  },
+  beforeDestroy() {
+    this.removeIdleListeners();
+  },
+  methods: {
+    trackInactivity() {
+      if (this.isInactive || this.isOpen) {
+        return;
+      }
+
+      this.clearAllTimeouts();
+
+      this.inactivityTimeoutId = setTimeout(() => {
+        this.isOpen = true;
+        this.startCountdown();
+
+        this.$modal.show('inactivityModal');
+      }, this.showModalAfter * 1000);
+    },
+    startCountdown() {
+      const endDate = new Date();
+
+      endDate.setSeconds(endDate.getSeconds() + this.courtesyCountdown);
+
+      const checkCountdown = () => {
+        const now = new Date();
+
+        if (now >= endDate) {
+          this.isInactive = true;
+          this.disablePulling();
+        } else {
+          this.courtesyCountdown = Math.floor((endDate - now) / 1000);
+          this.courtesyTimerId = setTimeout(checkCountdown, 1000);
+        }
+      };
+
+      checkCountdown();
+    },
+    addIdleListeners() {
+      document.addEventListener('mousemove', this.trackInactivity);
+      document.addEventListener('mousedown', this.trackInactivity);
+      document.addEventListener('keypress', this.trackInactivity);
+      document.addEventListener('touchmove', this.trackInactivity);
+      document.addEventListener('visibilitychange', this.trackInactivity);
+    },
+    removeIdleListeners() {
+      document.removeEventListener('mousemove', this.trackInactivity);
+      document.removeEventListener('mousedown', this.trackInactivity);
+      document.removeEventListener('keypress', this.trackInactivity);
+      document.removeEventListener('touchmove', this.trackInactivity);
+      document.addEventListener('visibilitychange', this.trackInactivity);
+    },
+
+    resume() {
+      this.isInactive = false;
+      this.courtesyCountdown = this.courtesyTimer;
+      this.clearAllTimeouts();
+      this.trackInactivity();
+
+      this.isOpen = false;
+      this.$modal.hide('inactivityModal');
+    },
+
+    refresh() {
+      window.location.reload();
+    },
+
+    disablePulling() {
+      this.$store.dispatch('rancher/unsubscribe');
```

Comment:
my advice here wasn't great. we should unsubscribe from all available stores (which might include `harvester`, `epinio`, extensions, etc). This can be done with the following

```
shell/components/Inactivity.vue
disablePulling() {
      this.$store.dispatch('unsubscribe');

shell/store/index.js
  unsubscribe( { state, dispatch }) {
    // It would be nice to grab all vuex module stores that we've registered, apparently this is only possible via the
    // internal properties store._modules.root._children.
    // So instead loop through all state entries to find stores
    return Object.entries(state).filter(([storeName, storeState]) => {
      if (storeState?.allowStreaming) {
        dispatch(`${ storeName }/unsubscribe`);
      }
    });
  }
```